File: Patterns/04. Right Triangle Patterns (Top-Right)/37. Right Triangle TR Binary/trb.cpp
#include <iostream>
using namespace std;
int main(){
    int n;
    cout<<"Enter a Number : ";
    cin>>n;

    for(int i=1; i<=n; i++){
        // print spaces
        for(int j=1; j<=n-i; j++){
            cout<< "  ";
        }
        // print stars
        for(int j=1; j<=i;j++){
            cout<<j%2<<" ";
        }
        cout << '\n';
    }
}

/* 

Output

        1 
      1 0 
    1 0 1 
  1 0 1 0 
1 0 1 0 1   

*/ 

File: Patterns/04. Right Triangle Patterns (Top-Right)/37. Right Triangle TR Binary/trb.java
import java.util.*;
public class trb{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a Number : ");
        int n = sc.nextInt();

        for(int i=1; i<=n; i++){
            // print spaces
            for(int j=1; j<=n-i; j++){
                System.out.print("  ");
            }
            // print stars
            for(int j=1; j<=i; j++){
                System.out.print(j%2 + " ");
            }
            System.out.println();
        }
        sc.close();
    }
}

/* 

Output

        1 
      1 0 
    1 0 1 
  1 0 1 0 
1 0 1 0 1   

*/
